src/coreclr/src/vm/eventpipe.cpp

```diff
@@ -169,10 +181,9 @@ void EventPipe::EnableViaEnvironmentVariables()
                 {
                     return;
                 }
-                // SampleProfiler can't be enabled on startup yet.
-                else if (wcscmp(W("Microsoft-DotNETCore-SampleProfiler"), configuration.GetProviderName()) == 0)
+                if (wcscmp(W("Microsoft-DotNETCore-SampleProfiler"), configuration.GetProviderName()) == 0)
                 {
-                    providerCnt -= 1;
```

Comment:
I think there is no longer a case below where providerCnt == 0, the if can be removed.